### ISSUE TITLE
-[SLButton isTappable] returns NO for a UITabBarItem element

### ISSUE THREAD

**Issue by nzhuk** [inkling/Subliminal#41]
I have a standard UITabBarController with a UITabBar and 4 UITabBarItems in it. When I query for a UITabBarItem with title "ButtonTitle", I get a valid result back and isValidAndVisible returns YES. However, isTappable returns NO and therefore I cannot tap it during the test.

``` objc
    SLButton *tabBarButton = UIAElement([SLButton elementWithAccessibilityLabel:@"ButtonTitle"]);
    SLAssertTrue([tabBarButton isValidAndVisible], @"Tab bar button missing.");
    SLAssertTrue([tabBarButton isTappable], @"Tab bar button isn't tappable.");
```

Normal UIButtons in the same view are working fine, and I can tap then normally with the 'tap' method. Any advice would be greatly appreciated.

**Comment by aegolden** [comment]
This does appear to be a bug in Subliminal.  I've pushed up a branch here:

https://github.com/inkling/Subliminal/tree/test_for_buttons_in_tab_bar

with integration test cases in the SLButton test suite that demonstrate this issue.

**Comment by wearhere** [comment]
Thanks for putting up test cases @aegolden.

What's going on is that Subliminal is not accounting for `UITabBar` being one of the classes that [forces its presence in the accessibility hierarchy](https://github.com/inkling/Subliminal/blob/master/Sources/Classes/UIAutomation/User%20Interface%20Elements/NSObject+SLAccessibilityHierarchy.m#L60) even if it's not accessible, so we're trying to identify the button directly, rather than being a button inside the tab bar (roughly, we're sending `target.frontMostApp().mainWindow().buttons()["Favorites"]` vs. `target.frontMostApp().mainWindow().tabBars()[0].buttons()["Favorites]`).

The fix is really simple, adding

``` objc
@implementation UITabBar (SLAccessibilityHierarchy)
- (BOOL)classForcesPresenceInAccessibilityHierarchy {
    return YES;
}
@end
```

to `NSObject+SLAccessibilityHierarchy.m`. We should add a test case to `SLElementMatchingTest` vs. `SLButtonTest` though because this is really a failure to properly match the button, nothing to do with tapping in particular. (On the test branch, `-[SLButtonTest testButtonsInTabBarsAreValidAndVisible]` succeeds because finding the element and assessing its visibility don't involve talking to UIAutomation).

**Comment by nzhuk** [comment]
Thanks guys, this seems to fix the issue!

**Comment by wearhere** [comment]
Sure @nzhuk. We'll have the fix in master once I fix up #42's build.
